content/asm_6.md

```diff
@@ -1,181 +1,544 @@
+
+### Data transfer instructions
+
+As we know from the [4th part](./asm_4.md) - data transfer instructions are used to move data between different locations. `x86_64` CPU provides special set of instructions for transfer of floating point data. If you are going to use the `x87 FPU` unit, one of the most common data transfer instructions are:
+
+- `fld` - Load floating point value.
+- `fst` - Store floating point value.
+
+These instructions are used by the `x87 FPU` unit to store and load the floating-point numbers at/from the stack.
+
+To work with `XMM` registers, the following data transfer instructions are supported by an `x86_64` CPU:
+
+- `movss` - Move single-precision floating-point value between the `XMM` registers or between an `XMM` register and memory,
+- `movsd` - Move double-precision floating-point value between the `XMM` registers or between an `XMM` register and memory.
+- `movhlps` - Move two packed single-precision floating-point values from the high quadword of an `XMM` register to the low quadword of another `XMM` register.
+- `movlhps` - Move two packed single-precision floating-point values from the low quadword of an `XMM` register to the high quadword of another `XMM` register.
+- And others.
+
+### Floating-point arithmetic instructions
 
-You can note two little differences here:
+The floating-point arithmetic instructions perform arithmetic operations such as add, subtract, multiplication, and division on single or double precision floating-point values. The following floating-point arithmetic instructions exist:
 
-* Section definition starts with . symbol
-* Main routine defines with .globl instead global as we do it in nasm
+- `addss` - Add single-precision floating-point values.
+- `addsd` - Add dobule-precision floating point values.
+- `subss` - Subtract single-precision floating-point values.
+- `subsd` - Subtract double-precision floating-point values.
+- `mulss` - Multiply single-precision floating-point values.
+- `mulsd` - Multiply double-precision floating-point values.
+- `divss` - Divide single-precision floating-point values. 
+- `divsd` - Divide double-precision floating-point values.
 
-Also gas uses another directives for data defintion:
+All of these instructions expects two operands to execute the given operation of addition, subtraction, multiplication, or division. After the operation will be executed, the result will be stored in the first operand.
+
+### Floating-Point control instructions
+
+As we already know, the main goal of this type of instructions is to manage the control flow of our programs. For the integer comparison we have seen the `cmp` instruction. But this instruction will not work with floating-point values. Although, the result of the comparison is controlled by the same `rflags` registers that we have seen in the [4th part](./asm_4.md#Control transfer instructions).
+
+The general instruction to compare floating-point numbers is `cmpss`.
```

Comment:
I'd join it with the previous paragraph, as it is closely related. Also... isn't any information missing? Seems like this section ended too suddenly.